write-fonts/src/pens.rs

```diff
@@ -0,0 +1,369 @@
+//! Pen implementations based on <https://github.com/fonttools/fonttools/tree/main/Lib/fontTools/pens>
+
+use font_types::{Pen, PenCommand};
+use kurbo::{Affine, BezPath, Point};
+
+/// A pen that transforms params using [kurbo::Affine].
+pub struct TransformPen<'a, T: Pen> {
+    inner_pen: &'a mut T,
+    transform: Affine,
+}
+
+impl<'a, T: Pen> TransformPen<'a, T> {
+    pub fn new(inner_pen: &'a mut T, transform: Affine) -> TransformPen<'a, T> {
+        TransformPen {
+            inner_pen,
+            transform,
+        }
+    }
+
+    fn map_point(&self, x: f32, y: f32) -> (f32, f32) {
+        let pt = self.transform
+            * Point {
+                x: x as f64,
+                y: y as f64,
+            };
+        (pt.x as f32, pt.y as f32)
```

Comment:
all this back and forth between f32 and f64 made me wonder.. are 32 bits enough for the pen protocol? Kurbo itself uses f64 for BezPath and Point. I guess it depends on the circumstances how much precision one needs. Perhaps the font-types Pen trait could be made generic over the coodinate float type. Or maybe I'm fine like that.

Reply:
These types evolved separately and the primary purpose for the `Pen` trait (originally called `PathSink`) was to support callback based output from the glyph loader. Incidentally, the original trait was generic over coordinate type.

Reply:
I would like to tackle any such modification to the Pen trait outside this PR